pkg/apis/pipeline/v1/pipeline_types_test.go

```diff
@@ -644,6 +644,46 @@ func TestPipelineTask_validateMatrix(t *testing.T) {
 					Name: "barfoo", Value: ParamValue{Type: ParamTypeArray, ArrayVal: []string{"bar", "foo"}},
 				}}},
 		},
+	}, {
+		name: "parameters in include matrix are strings",
+		pt: &PipelineTask{
+			Name: "task",
+			Matrix: &Matrix{
+				Include: []MatrixInclude{{
+					Name: "build-1",
+					Params: []Param{{
+						Name: "IMAGE", Value: ParamValue{Type: ParamTypeString, StringVal: "image-1"},
+					}, {
+						Name: "DOCKERFILE", Value: ParamValue{Type: ParamTypeString, StringVal: "path/to/Dockerfile1"},
+					}}},
+				}},
+		},
+	}, {
+		name: "parameters in include matrix are arrays",
+		pt: &PipelineTask{
+			Name: "task",
+			Matrix: &Matrix{
+				Include: []MatrixInclude{{
+					Name: "build-1",
+					Params: []Param{{
+						Name: "foobar", Value: ParamValue{Type: ParamTypeArray, ArrayVal: []string{"foo", "bar"}},
+					}, {
+						Name: "barfoo", Value: ParamValue{Type: ParamTypeArray, ArrayVal: []string{"bar", "foo"}}}},
+				}}},
+		},
+		wantErrs: &apis.FieldError{
+			Message: "invalid value: parameters of type string only are allowed in matrix",
+			Paths:   []string{"matrix[barfoo]", "matrix[foobar]"},
```

Comment:
this is why I suggested that we fix the field keys, the paths should be:
```suggestion
			Paths:   []string{"matrix.include.params[barfoo]", "matrix.include.params[foobar]"},
```

Reply:
and the same applies for the other test cases